integrations/gcp/main.py

```diff
@@ -0,0 +1,111 @@
+import http
+
+from fastapi import Request, Response
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from gcp_core.feed_event import (
+    AssetHasNoProjectAncestorError,
+    GotFeedCreatedSuccessfullyMessage,
+    feed_event_to_resource,
+    get_project_from_ancestors,
+    parse_asset_data,
+)
+from gcp_core.search.iterators import iterate_per_available_project
+from gcp_core.search.searches import (
+    ResourceNotFoundError,
+    list_all_topics_per_project,
+    search_all_folders,
+    search_all_organizations,
+    search_all_projects,
+    search_all_resources,
+)
+from gcp_core.search.utils import (
+    AssetTypesWithSpecialHandling,
+)
+
+
+@ocean.on_resync(kind=AssetTypesWithSpecialHandling.FOLDER)
+async def resync_folders(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    with logger.contextualize(kind=AssetTypesWithSpecialHandling.FOLDER):
```

Comment:
i think we already have the kind logging
update if im wrong

Reply:
we do